### ISSUE TITLE
Proposal: KaiaContrib

### ISSUE THREAD

**Issue by Rushabt11** [kaiachain/kaia-dapp-mono#336]
Description
- The Github Bounty Dispenser is a blockchain-based platform that allows users to link their Github accounts with Kaia wallets to participate in open-source projects and earn bounties. Project maintainers can approve bounties and reward contributors. The platform includes a dashboard for tracking user profiles, managing bounties, and displaying a leaderboard of contributors based on bounty earnings. Built on the Kaia Blockchain, this project aims to make open-source contributions more accessible, transparent, and rewarding.
- Work Flow - link your Github account with your Kaia wallet address to participate in bounties for open-source contributions.
Bounty track, Once a bounty is approved by the project maintainer, contributors are rewarded automatically and securely through the Kaia blockchain. Transactions are transparent and tamper-proof, ensuring fair compensation for all. Users can track their contributions and earnings, while maintainers can manage open bounties, approve work, and monitor project progress. The dashboard also provides a leaderboard, showcasing top contributors and their total earnings.

- Surf Szn

**Comment by Rushabt11** [comment]
@zxstim update??

**Comment by zxstim** [comment]
Do you link in backend?

**Comment by 0xTaneja** [comment]
@zxstim which backend ??

**Comment by Rushabt11** [comment]
@zxstim What do you mean ??

**Comment by zxstim** [comment]
> link your Github account with your Kaia wallet address to participate in bounties for open-source contributions.

this. Do you use a backend to link?

I prefer a cleaner approach like this. Contributor created a repo with your designated name (something very unique), in the repo, there is a json file with a specific format containing the payout address. This way, you can call the github API to fetch the address easily and user is responsible for making sure the address is correct.

**Comment by Rushabt11** [comment]
@zxstim yeah this is too possible but don't you think Contributors might prefer a more private method of linking their wallet address, as GitHub repositories are public by default unless specifically made private and also If the naming convention or JSON Structure isn’t strictly followed, this could increase complexity for user as well as contributor.

**Comment by zxstim** [comment]
You are paying out from a public wallet anyway because the prize pool should be public for people to verify that you have money to pay.

From there, it's easy to link just by matching the amount and timing.

It's normal for grant to be public because it's also a positive signal for others that people are getting paid well.

Technically i want to go even further by having the submission to be onchain entirely and just pay to the address that submitted (submisison info in calldata) but i can live with a couple things offchain for now. I don't like other people handling addresses in db because they might do a bad job with the api and mix up the pay out addresses, then throw blame and make accusations. Or they happen to drop the db and make everyone to link again. Or you have to account for re-linking because my previous address is compromised for example.

**Comment by zxstim** [comment]
And btw, for this program, you are supposed to submit a Dorahacks BUIDL with your address pasted in the BUIDL content. Been doing that and people have no issue.

If you don't want people to know who you are, use a mixer and mix it before depositing into a cex.

**Comment by Rushabt11** [comment]
@zxstim okay fine lets have json approach then ! I will start to make things easier and cleaner as this is mvp.

**Comment by zxstim** [comment]
use a mix of tag system, github bot to track status.

Check this Issue tab to see the way i manage the payout. There are stages.
